File: src/main/java/com/jrmplugin/core/SafeJTextArea.java
package com.jrmplugin.core;

import javax.swing.*;

public class SafeJTextArea extends JTextArea {

    private static final Object OBJECT = new Object();

    public SafeJTextArea(int rows, int columns) {
        super(rows, columns);
    }

    @Override
    public void setText(String t) {
        synchronized (OBJECT) {
            super.setText(t);
        }
    }
}

File: src/main/java/com/jrmplugin/core/SafeJTextArea.kt
package com.jrmplugin.core

import javax.swing.JTextArea

class SafeJTextArea(rows: Int, columns: Int) : JTextArea(rows, columns) {
    override fun setText(t: String) {
        synchronized(OBJECT) { super.setText(t) }
    }

    companion object {
        private val OBJECT = Any()
    }
}